server/plugin/webhook.go

```diff
@@ -1377,3 +1385,44 @@ func (p *Plugin) postStarEvent(event *github.StarEvent) {
 		}
 	}
 }
+
+func (p *Plugin) postWorkflowJobEvent(event *github.WorkflowJobEvent) {
+	if event.GetAction() != actionCompleted {
+		return
+	}
+
+	// Create a post only when the workflow job is completed and fails
+	if event.GetWorkflowJob().GetConclusion() != workflowJobFail {
+		return
+	}
+
+	repo := event.GetRepo()
+	subs := p.GetSubscribedChannelsForRepository(repo)
+
+	if len(subs) == 0 {
+		return
+	}
+
+	newWorkflowJobMessage, err := renderTemplate("newWorkflowJob", event)
+	if err != nil {
+		p.client.Log.Warn("Failed to render template", "Error", err.Error())
+		return
+	}
+
+	for _, sub := range subs {
+		if !sub.Workflows() {
+			continue
+		}
+
+		post := &model.Post{
+			UserId:    p.BotUserID,
+			Type:      "custom_git_workflow_job",
+			Message:   newWorkflowJobMessage,
+			ChannelId: sub.ChannelID,
+		}
```

Comment:
Multiple subs should not share a post, see https://github.com/mattermost/mattermost-plugin-github/pull/755